### PR TITLE
test/e2e: several fixes

### PR DESCRIPTION
**Pull request by Luap99**
see commits

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
Really nice improvements, LGTM

> [FAILED] DeferCleanup callback returned error: unlinkat /tmp/ginkgo349969710/root/vfs/dir/b993cfcfd8fdc2731b308222fce8b5ce7a681780ab276b6a23e1b0fe3ce06bbb/bin: permission denied

Not yet sure where that's coming from.

**Comment by Luap99** [comment]
> Really nice improvements, LGTM
> 
> > [FAILED] DeferCleanup callback returned error: unlinkat /tmp/ginkgo349969710/root/vfs/dir/b993cfcfd8fdc2731b308222fce8b5ce7a681780ab276b6a23e1b0fe3ce06bbb/bin: permission denied
> 
> Not yet sure where that's coming from.

That is the classic problem that as rootless you cannot delete files owned by subuid/gid without joining the userns. We have to use `podman unshare rm -rf` in that case and I thought I covered that, running locally passed because it used the images from my local cache so I missed that one

**Comment by vrothberg** [comment]
> That is the classic problem that as rootless you cannot delete files owned by subuid/gid without joining the userns.

That I see but I fail to see where this was removed in the PR.

File: test/e2e/common_test.go [hunk 1]
```diff
@@ -140,7 +146,7 @@ var _ = SynchronizedBeforeSuite(func() []byte {
 	// Cache images
 	cwd, _ := os.Getwd()
 	INTEGRATION_ROOT = filepath.Join(cwd, "../../")
-	podman := PodmanTestSetup(os.TempDir())
+	podman := PodmanTestSetup(GinkgoT().TempDir())
```
Comment (Luap99) [inline test/e2e/common_test.go:149]: right here

Comment (vrothberg) [reply test/e2e/common_test.go:149]: That I saw but I fail to understand why worked before.  We did a `RemoveAll` before that is failing now.

I don't want to bother much but am trying to put the puzzle in my head together.

**Review by vrothberg — COMMENTED** [review]
@giuseppe @flouthoc PTAL

File: test/e2e/common_test.go (hunk 1 above)
Comment (Luap99) [reply test/e2e/common_test.go:149]: No we did never call RemoveAll() on that path. This path right here was always leaked, just check /tmp for root, runroot directories. And yes RemoveAll() will work but only as long as you do not pull images with uid > 0.

**Review by flouthoc — APPROVED** [review]
LGTM
/approve
/hold

File: test/e2e/common_test.go (hunk 1 above)
Comment (vrothberg) [reply test/e2e/common_test.go:149]: D'oh!   Thanks!

**Comment by edsantiago** [comment]
/hold

I need time to understand the `ImageCache` changes. My home network is ultra-slow, and if I have to re-pull images each time, I will never be able to run ginkgo again

**Comment by Luap99** [comment]
> I need time to understand the ImageCache changes. My home network is ultra-slow, and if I have to re-pull images each time, I will never be able to run ginkgo again

You don't have to repull, ImageCache is the path were the extracted images are stored not the cached tar files. That location is still the same.

**Comment by edsantiago** [comment]
>You don't have to repull, [...] cached tar files. That location is still the same.

Confirmed. Thank you. Thank you for doing this, and extra thanks for fixing the `TMPDIR/root` and `TMPDIR/runroot` bug, that one always annoyed me!

All other commits LGTM, and I've verified that timings in run logs look reasonable (at a glance).

/hold cancel

**Comment by edsantiago** [comment]
Uh-oh. @Luap99 I didn't notice this when I reviewed logs, but my flake logger did: [this PR flaked](https://api.cirrus-ci.com/v1/artifact/task/5642322990333952/html/int-podman-fedora-38-root-host-boltdb.log.html), in fetching cache images, NBD that happens all the time because of flaky quay. However:
```
[+0203s] [SynchronizedAfterSuite] [FAILED] [0.052 seconds]
...
           [FAILED] Unexpected error:
               <*fs.PathError | 0xc0004c5e90>: 
               open /timings-1: no such file or directory
               {
                   Op: "open",
                   Path: "/timings-1",
                   Err: <syscall.Errno>0x2,
...
The `/timings-1` is once again missing `LockTmpDir`.

**Comment by Luap99** [comment]
> Uh-oh. @Luap99 I didn't notice this when I reviewed logs, but my flake logger did: [this PR flaked](https://api.cirrus-ci.com/v1/artifact/task/5642322990333952/html/int-podman-fedora-38-root-host-boltdb.log.html), in fetching cache images, NBD that happens all the time because of flaky quay. However:
> 
> ```
> [+0203s] [SynchronizedAfterSuite] [FAILED] [0.052 seconds]
> ...
>            [FAILED] Unexpected error:
>                <*fs.PathError | 0xc0004c5e90>: 
>                open /timings-1: no such file or directory
>                {
>                    Op: "open",
>                    Path: "/timings-1",
>                    Err: <syscall.Errno>0x2,
> ```
> 
> The `/timings-1` is once again missing `LockTmpDir`.

Yes that is fine and the failure mode we had locally before. SynchronizedAfterSuite will run even when SynchronizedBeforeSuite fails. Which means we can end up with an unset LockTmpDir. But in this case we only open for reading so it should no longer create the files.

**Pull request merged**